apps/tx-builder/src/components/Builder.tsx

```diff
@@ -0,0 +1,319 @@
+import React, { useState, useEffect, ReactElement, useCallback } from 'react';
+import {
+  Button,
+  Text,
+  Title,
+  TextField,
+  GenericModal,
+  Select,
+  ModalFooterConfirmation,
+  ButtonLink,
+} from '@gnosis.pm/safe-react-components';
+import { useSafeAppsSDK } from '@gnosis.pm/safe-apps-react-sdk';
+import styled from 'styled-components';
+import { AbiItem } from 'web3-utils';
+
+import { ContractInterface } from '../hooks/useServices/interfaceRepository';
+import useServices from '../hooks/useServices';
+import { ProposedTransaction } from '../typings/models';
+import { ModalBody } from './ModalBody';
+import { Examples } from './Examples';
+
+const ButtonContainer = styled.div`
+  display: flex;
+  justify-content: space-between;
+  margin-top: 15px;
+`;
+
+const StyledTextField = styled(TextField)`
+  && {
+    width: 520px;
+    margin-bottom: 10px;
+  }
+`;
+
+const StyledSelect = styled(Select)`
+  width: 520px;
+`;
+
+const StyledExamples = styled.div`
+  button {
+    padding: 0;
+  }
+`;
+
+type Props = {
+  contract: ContractInterface | undefined;
+  to: string | null;
+};
+
+export const Builder = ({ contract, to }: Props): ReactElement | null => {
+  const { sdk, safe } = useSafeAppsSDK();
+  const services = useServices(safe.network);
+
+  const [toInput, setToInput] = useState('');
+  const [valueInput, setValueInput] = useState('');
+
+  const [transactions, setTransactions] = useState<ProposedTransaction[]>([]);
+  const [reviewing, setReviewing] = useState(false);
+  const [selectedMethodIndex, setSelectedMethodIndex] = useState(0);
+  const [showExamples, setShowExamples] = useState(false);
+  const [addTxError, setAddTxError] = useState<string | undefined>();
+  const [inputCache, setInputCache] = useState<string[]>([]);
+  const [isValueInputVisible, setIsValueInputVisible] = useState(false);
+
+  const handleMethod = useCallback(
+    async (methodIndex: number) => {
+      if (!contract || contract.methods.length <= methodIndex) return;
+      setSelectedMethodIndex(methodIndex);
+    },
+    [contract],
+  );
+
+  const handleInput = useCallback(
+    async (inputIndex: number, input: string) => {
+      inputCache[inputIndex] = input;
+      setInputCache(inputCache.slice());
+    },
+    [inputCache],
+  );
+
+  const getInputHelper = (input: any) => {
```

Comment:
Why do we not use any react hooks here? (E.g. useCallback). Also why is this `any`?